backend/infrahub/task_manager/event.py

```diff
@@ -0,0 +1,93 @@
+import uuid
+from typing import Any
+
+from prefect.client.orchestration import PrefectClient, get_client
+from prefect.events.filters import EventFilter, EventIDFilter, EventNameFilter, EventRelatedFilter
+from prefect.events.schemas.events import Event as PrefectEventModel
+from prefect.events.schemas.events import ResourceSpecification
+from pydantic import TypeAdapter
+
+from infrahub.log import get_logger
+from infrahub.utils import get_nested_dict
+
+log = get_logger()
+
+# pylint: disable=unused-argument
+
+
+class PrefectEventData(PrefectEventModel):
+    def get_branch(self) -> str | None:
+        for resource in self.related:
+            if resource.get("prefect.resource.id") != "infrahub.branch":
+                continue
+            if "prefect.resource.name" not in resource:
+                continue
+            return resource.get("prefect.resource.name")
+        return None
+
+
+class PrefectEvent:
+    @classmethod
+    async def query_events(cls, client: PrefectClient, filters: EventFilter | None = None) -> list[PrefectEventData]:
+        body = {}
+        if filters:
+            body["filter"] = filters.model_dump(mode="json", exclude_unset=True)
+
+        response = await client._client.post("/events/filter", json=body)
+        response.raise_for_status()
```

Comment:
Doesn't have to be in this PR but I think we should perhaps consider having some abstractions around `PrefectClient` so that the resolvers don't need to care about any prefect internals.